### ISSUE TITLE
Don’t Panic, Return Err

### ISSUE THREAD

**Issue by ryancurrah** [cristalhq/hedgedhttp#23]
I’m interested in using this lib but have concerns over the use of panic in it.

Just curious as to why it’s used?

If it was an oversight would you’d be open to a PR that returns an error instead?

Thank you.

**Comment by cristaloleg** [comment]
Hi, is your question about params validation?

If so - there is nothing special, so xan be changed.

**Comment by cristaloleg** [comment]
```
func NewClient(timeout time.Duration, upto int, client *http.Client) (*http.Client, error) {
func NewClientAndStats(timeout time.Duration, upto int, client *http.Client) (*http.Client, *Stats, error) {
func NewRoundTripper(timeout time.Duration, upto int, rt http.RoundTripper) (http.RoundTripper, error) {
func NewRoundTripperAndStats(timeout time.Duration, upto int, rt http.RoundTripper) (http.RoundTripper, *Stats, error) {
```

Looks neat for me. Feel free to pick it up ;)

**Comment by ryancurrah** [comment]
Will do.

**Comment by cristaloleg** [comment]
Fixed in #24
